framework/src/org/checkerframework/common/signatureinference/SignatureInferenceScenes.java

```diff
@@ -0,0 +1,726 @@
+                    shouldIgnore = true;
+                }
+            }
+        }
+        DefaultFor defaultQualForLocation = am.getAnnotationType().asElement().
+                getAnnotation(DefaultFor.class);
+        if (!shouldIgnore && defaultQualForLocation != null) {
+            for (DefaultLocation loc : defaultQualForLocation.value()) {
+                if (loc == DefaultLocation.ALL || loc == location) {
+                    shouldIgnore = true;
+                }
+            }
+        }
+
+        return shouldIgnore;
+    }
+
+    /**
+     * Returns a subset of annosSet, consisting of the annotations supported
+     * by atf.
+     */
+    private static Set<Annotation> getSupportedAnnosInSet(Set<Annotation> annosSet,
+            AnnotatedTypeFactory atf) {
+        Set<Annotation> output = new HashSet<>();
+        Set<Class<? extends java.lang.annotation.Annotation>> supportedAnnos
+                = atf.getSupportedTypeQualifiers();
+        for (Annotation anno: annosSet) {
+            for (Class<? extends java.lang.annotation.Annotation> clazz : supportedAnnos) {
+                // TODO: Remove comparison by name, and make this routine more efficient.
+                if (clazz.getName().equals(anno.def.name)) {
+                    output.add(anno);
+                }
+            }
+        }
+        return output;
+    }
+
+    /**
+     * Returns an {@link org.checkerframework.framework.type.AnnotatedTypeMirror}
+     * containing the {@link annotations.Annotation}s of an
+     * {@link annotations.el.ATypeElement}.
+     * @param type the {@link annotations.el.ATypeElement}.
+     * @param atf the annotated type factory of a given type system, whose
+     * type hierarchy will be used.
+     * @param tm the underlying type.
+     */
+    private static AnnotatedTypeMirror typeElementToATM(
+            ATypeElement type, AnnotatedTypeFactory atf, TypeMirror tm) {
+        AnnotatedTypeMirror atm = AnnotatedTypeMirror.createType(tm, atf,
+                false);
+        Set<Annotation> annos = getSupportedAnnosInSet(type.tlAnnotationsHere,
+                atf);
+        for (Annotation anno: annos) {
+            AnnotationMirror am = annotationToAnnotationMirror(
+                    anno, atf.getProcessingEnv());
+            atm.addAnnotation(am);
+        }
+        if (tm.getKind() == TypeKind.ARRAY) {
+            AnnotatedArrayType aat = (AnnotatedArrayType) atm;
+            for (ATypeElement innerType : type.innerTypes.values()) {
+                TypeMirror at = aat.getUnderlyingType().getComponentType();
+                aat.setComponentType(typeElementToATM(innerType, atf,at));
+            }
+        }
+        return atm;
+    }
+
+   /**
+    * Updates an {@annotations.el.ATypeElement} to have the annotations of an
+    * {@link org.checkerframework.framework.type.AnnotatedTypeMirror} passed
+    * as argument. Annotations in the original set that are an implementation
```

Comment:
It's an annotation that cannot be written in source code.  The documentation should be more clear.